modules/catalog/src/main/java/org/apache/ignite/internal/catalog/CatalogManagerImpl.java

```diff
@@ -321,8 +328,10 @@ public CompletableFuture<Void> execute(List<CatalogCommand> commands) {
     }
 
     private void registerCatalog(Catalog newCatalog) {
-        catalogByVer.put(newCatalog.version(), newCatalog);
-        catalogByTs.put(newCatalog.time(), newCatalog);
+        inBusyLock(busyLock, () -> {
+            catalogByVer.put(newCatalog.version(), newCatalog);
+            catalogByTs.put(newCatalog.time(), newCatalog);
+        });
```

Comment:
This method is called in `OnUpdateHandlerImpl.handle(...)`.
` handleFutures.add(onUpdateHandler.handle(update, event.timestamp(), event.revision()));`
It make sense to guard the `UpdateListener.onUpdate(WatchEvent)` with busylock instead.

Reply:
seems no need to use busyLock here it correctly stops before.